lib/client/api.go

```diff
@@ -834,6 +875,15 @@ func (tc *TeleportClient) startPortForwarding(nodeClient *NodeClient) error {
 			go nodeClient.listenAndForward(socket, net.JoinHostPort(fp.DestHost, strconv.Itoa(fp.DestPort)))
 		}
 	}
+	if len(tc.Config.DynamicForwardedPorts) > 0 {
+		for _, fp := range tc.Config.DynamicForwardedPorts {
+			socket, err := net.Listen("tcp", net.JoinHostPort(fp.SrcIP, strconv.Itoa(fp.SrcPort)))
```

Comment:
if you created 3 listeners and failed to bind on 4th listener, remaining 3 listeners will linger and leak? Can you make sure the code closes the listeners on client close or error return (I would use contexts)

Reply:
I think a better UX would be to just log that we failed to bind to a particular address but continue to bind to the ones that we can.

What do you think?